### PR TITLE
fix: remove duplicate loan query, use async await to control data and…

### PR DESCRIPTION
**Pull request by mcstover**
… loan loading sequence.

### CODE REVIEW

File: src/components/CorporateCampaign/LoanSearch/LoanSearchFilters.vue [hunk 1]
```diff
@@ -352,7 +352,6 @@ export default {
 				return !this.excludedTags.includes(tag.name);
 			});
 		});
-		this.setLoanDisplayMode(this.activeLoanDisplay);
```
Comment (mcstover) [inline src/components/CorporateCampaign/LoanSearch/LoanSearchFilters.vue:355]: This was triggering the `showLoans` mechanisms immediately upon page load. It the flows into the loan display and fired a default loan query before we knew anything about the promotion resulting in potentially ineligible loans to show on the page.

File: src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue [hunk 2]
```diff
@@ -698,8 +698,12 @@ export default {
 		// const basketItems = this.rawPageData?.shop?.basket?.items?.values ?? [];
 		this.loadingPage = basketItems.some(item => item.__typename === 'LoanReservation'); // eslint-disable-line no-underscore-dangle, max-len
 	},
-	mounted() {
+	async mounted() {
 		this.$root.$on('jumpToLoans', this.jumpToLoans);
+
+		// check for applied promo
+		await this.verifyOrApplyPromotion();
```
Comment (mcstover) [inline src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue:705]: This method starts our data aggregation for the promo state, we want to control this flow deliberately so we have all the promo information before proceeding with other steps.

File: src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue [hunk 3]
```diff
@@ -985,35 +986,35 @@ export default {
 		},
 	},
 	methods: {
-		verifyOrApplyPromotion() {
+		async verifyOrApplyPromotion() {
```
Comment (mcstover) [inline src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue:989]: Using the async/await flow here will ensure we awat the promo application or getting promo info from the basket before any other steps.

File: src/components/CorporateCampaign/LoanSearch/LoanSearchFilters.vue (hunk 1 above)
Comment (mcstover) [reply src/components/CorporateCampaign/LoanSearch/LoanSearchFilters.vue:355]: I've verified that the default `rows` view and the contentful setting that swaps to `grid` both work and load up properly.

**Review by dyersituations — APPROVED** [review]

**Review by michelleinez — APPROVED** [review]

**Pull request merged**